lib/glucose-get-last.js

```diff
@@ -1,329 +1,91 @@
-    if (! digits) { digits = 0; }
-    var scale = Math.pow(10, digits);
-    return Math.round(value * scale) / scale;
-}
+  var getLastGlucose = function (data) {
+      data = data.filter(function(obj) {
+        return obj.glucose || obj.sgv;
+      }).map(function prepGlucose (obj) {
+          //Support the NS sgv field to avoid having to convert in a custom way
+          obj.glucose = obj.glucose || obj.sgv;
+          if ( obj.glucose !== null ) {
+              return obj;
+          }
+      });
 
-var getLastGlucose = function (data) {
-    data = data.filter(function(obj) {
-    return obj.glucose || obj.sgv;
-    }).map(function prepGlucose (obj) {
-        //Support the NS sgv field to avoid having to convert in a custom way
-        obj.glucose = obj.glucose || obj.sgv;
-        if ( obj.glucose !== null ) {
-            return obj;
-        }
-    });
+      var now = data[0];
+      var now_date = getDateFromEntry(now);
+      var change;
+      var last_deltas = [];
+      var short_deltas = [];
+      var long_deltas = [];
+      var last_cal = 0;
 
-    var now = data[0];
-    var now_date = getDateFromEntry(now);
-    var change;
-    var last_deltas = [];
-    var short_deltas = [];
-    var long_deltas = [];
-    var last_cal = 0;
+      //console.error(now.glucose);
+      for (var i=1; i < data.length; i++) {
+          // if we come across a cal record, don't process any older SGVs
+          if (typeof data[i] !== 'undefined' && data[i].type === "cal") {
+              last_cal = i;
+              break;
+          }
+          // only use data from the same device as the most recent BG data point
+          if (typeof data[i] !== 'undefined' && data[i].glucose > 38 && data[i].device === now.device) {
+              var then = data[i];
+              var then_date = getDateFromEntry(then);
+              var avgdelta = 0;
+              var minutesago;
+              if (typeof then_date !== 'undefined' && typeof now_date !== 'undefined') {
+                  minutesago = Math.round( (now_date - then_date) / (1000 * 60) );
+                  // multiply by 5 to get the same units as delta, i.e. mg/dL/5m
+                  change = now.glucose - then.glucose;
+                  avgdelta = change/minutesago * 5;
+              } else { console.error("Error: date field not found: cannot calculate avgdelta"); }
+              //if (i < 5) {
+                  //console.error(then.glucose, minutesago, avgdelta);
+              //}
+              // use the average of all data points in the last 2.5m for all further "now" calculations
+              if (-2 < minutesago && minutesago < 2.5) {
+                  now.glucose = ( now.glucose + then.glucose ) / 2;
+                  now_date = ( now_date + then_date ) / 2;
+                  //console.error(then.glucose, now.glucose);
+              // short_deltas are calculated from everything ~5-15 minutes ago
+              } else if (2.5 < minutesago && minutesago < 17.5) {
```

Comment:
if `minutesago` is exactly 2.5 will this cause issues? I would imagine that one of these comparisons includes an `=`